jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -89,108 +89,159 @@ var module = module;
 
 
   /**
-     Environment object for jujulib.
+    JEM object for jujulib.
 
-     Provides access to the JEM API.
-   */
+    Provides access to the JEM API.
+  */
 
   /**
-     Initializer
-
-     @function environment
-     @param url {String} The URL, including scheme and port, of the JEM instance.
-     @param bakery {Object} A bakery object for communicating with the JEM instance.
-     @returns {Object} A client object for making JEM API calls.
-   */
-  function environment(url, apiVersion, bakery) {
+    Initializer
+
+    @function jem
+    @param url {String} The URL of the JEM instance, including scheme, port
+      and the trailing slash, and excluding the API version.
+    @param bakery {Object} A bakery object for communicating with the JEM
+      instance.
+    @returns {Object} A client object for making JEM API calls.
+  */
+  function jem(url, bakery) {
     // XXX j.c.sackett 2016-03-16 We should probably adopt the generate query
     // mechanism from charmstore but that's a larger rewrite. For now we're
-    // making sure we can take the same initialization data for the jem url as
+    // making sure we can take the same initialization data for the JEM URL as
     // we do for the charmstore url.
-    this.jemURL = url + apiVersion;
+    this.url = url + 'v2';
```

Comment:
Because it makes no sense to accept that from the user, as we only support v2 and don't want to implement a client supporting multiple versions. So, to avoid confusion, we no longer accept API versions for jem and for the charm store, and we just define it here what's being used.